src/ixdat/measurements.py

```diff
@@ -264,26 +398,291 @@ def get_t_and_v(self, item, tspan=None):
+
+        new_measurement = self.__class__.from_dict(obj_as_dict)
+        return new_measurement
+
+    def select_value(self, *args, **kwargs):
+        """Return a new Measurement with the time(s) meeting criteria.
+
+        Can only take one arg or kwarg!
+        The `series_name` is `self.sel_str` if given an arg, kw if given a kwarg.
+        Either way the argument is the `value` to be selected for.
+
+        The method finds all time intervals for which `self[series_name] == value`
+        It then cuts the measurement according to each time interval and adds these
+        segments together. TODO: This can be done better, i.e. without chopping series.
+
+        TODO: greater-than and less-than kwargs.
+            Ideally you should be able to say e.g., `select(cycle=1, 0.5<potential<1)`
+        """
+        if len(args) >= 1:
+            if not self.sel_str:
+                raise BuildError(
+                    f"{self} does not have a default selection string "
+                    f"(Measurement.sel_str), and so selection only works with kwargs."
+                )
+            kwargs[self.sel_str] = args
+        if len(kwargs) > 1:
+            raise BuildError(
+                f"select_value got kwargs={kwargs} but can only be used for one value "
+                f"at a time. Use select_values for more."
+            )
+        new_measurement = self
+        ((series_name, value),) = kwargs.items()
+
+        t, v = self.grab(series_name)
+        mask = v == value  # linter doesn't realize this is a np array
+        mask_prev = np.append(False, mask[:-1])
+        mask_next = np.append(mask[1:], False)
+        interval_starts_here = np.logical_and(
+            np.logical_not(mask_prev), mask
+        )  # True at [0] if mask[0] is True.
+        interval_ends_here = np.logical_and(
+            mask, np.logical_not(mask_next)
+        )  # True at [-1] if mask[-1] is True.
+        t_starts = list(t[interval_starts_here])
+        t_ends = list(t[interval_ends_here])
+        tspans = zip(t_starts, t_ends)
+        meas = None
+        for tspan in tspans:
+            if meas:
+                meas = meas + new_measurement.cut(tspan)
+            else:
+                meas = new_measurement.cut(tspan)
+        new_measurement = meas
+        return new_measurement
+
+    def select_values(self, *args, **kwargs):
+        """Return a new Measurement with the time(s) in the measurement meeting criteria
+
+        Any series can be selected for using the series name as a key-word. Arguments
+        can be single acceptable values or lists of acceptable values. In the latter
+        case, each acceptable value is selected for on its own and the resulting
+        measurements added together.
+        # FIXME: That is sloppy because it mutliplies the number of DataSeries
+            containing the same amount of data.
+        If no key-word is given, the series name is assumed to
```

Comment:
docstring fixed. 
If this is called with args and also with `selector_name` as a kwarg, it will be as if just the most restrictive of those two was used. No error needed, I think.